Core/clim-core/presentation-translators.lisp

```diff
@@ -450,46 +450,37 @@ and used to ensure that presentation-translators-caches are up to date.")
-    (let* ((from-type (from-type translator)))
-      (unless (match-gesture (gesture translator) event modifier-state)
-        (return-from test-presentation-translator nil))
-      (unless (or (null (decode-parameters from-type))
-                  (presentation-typep (presentation-object presentation)
-                                      from-type))
-        (return-from test-presentation-translator nil))
-      (unless (or (null (tester translator))
-                  (funcall (tester translator) (presentation-object presentation)
-                           :presentation presentation :context-type context-type
-                           :frame frame :window window :x x :y y :event event))
-        (return-from test-presentation-translator nil))
-      (unless (or (tester-definitive translator)
-                  (null (decode-parameters context-type))
-                  (presentation-typep (call-presentation-translator
-                                       translator
-                                       presentation
-                                       context-type
-                                       frame
-                                       event
-                                       window
-                                       x y)
-                                      context-type))
-        (return-from test-presentation-translator nil))))
-  t)
+           (or (eq gesture t)
+               for-menu
+               (loop
+                  with modifiers = (if event
+                                       (event-modifier-state event)
+                                       modifier-state)
+                  for g in gesture
+                  thereis (and (eql modifiers (caddr g))
+                               (or (and button (eql button (cadr g)))
+                                   (and (null button)
+                                        (or (null event)
+                                            (eql (pointer-event-button
+                                                  event)
+                                                 (cadr g))))))))))
+    (and (match-gesture (gesture translator) event modifier-state)
+         (or (null (decode-parameters from-type))
+             (presentation-typep (presentation-object presentation) from-type))
+         (or (null (tester translator))
+             (funcall (tester translator) (presentation-object presentation)
+                      :presentation presentation :context-type context-type
+                      :frame frame :window window :x x :y y :event event))
+         (or (tester-definitive translator)
+             (null (decode-parameters context-type))
+             (presentation-typep
+              (call-presentation-translator translator presentation context-type
+                                            frame event window x y)
+              context-type))
+         t)))
```

Comment:
The final `t` should not be needed.

Reply:
original code returned either NIL or T, if we skip T then it will return a generalized boolean. It makes sense and does not contradict the spec, it is just that this change doesn't change function semantics. Should I change it?

Reply:
No, it's fine.